src/model/document.ts

```diff
@@ -238,14 +238,15 @@ export class Document implements Feature, Queryable {
   getByKind(kind: string, options?: QueryOptions): Set<Feature>;
   getByKind(kind: string, options?: QueryOptions): Set<Feature> {
     options = options || {};
-    if (this._featuresByKind && options.imported) {
+    if (this._featuresByKind && options.imported && !options.strictImports) {
```

Comment:
`!options.strictImports` is used in both checks below, can you pull it out of both and handle it separately?